x/auth/types/stdsignmsg.go

```diff
@@ -5,31 +5,32 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// StdSignMsg is a convenience structure for passing along
-// a Msg with the other requirements for a StdSignDoc before
-// it is signed. For use in the CLI.
+var _ types.UnpackInterfacesMessage = StdSignMsg{}
+
+// StdSignMsg is a convenience structure for passing along a Msg with the other
+// requirements for a StdSignDoc before it is signed. For use in the CLI.
 type StdSignMsg struct {
 	ChainID       string    `json:"chain_id" yaml:"chain_id"`
 	AccountNumber uint64    `json:"account_number" yaml:"account_number"`
 	Sequence      uint64    `json:"sequence" yaml:"sequence"`
+	TimeoutHeight uint64    `json:"timeout_height" yaml:"timeout_height"`
```

Comment:
Why will it be broken?

Reply:
It's required in a `StdTx`.

Reply:
It was never required in `StdTx` before. This requires everyone to change their signing algorithm. We are trying to deprecate `StdTx` not force everyone to change legacy signing and adopt new signing. I might even prefer to leave it off `StdTx` altogether, but if not `omitempty` is necessary to prevent breakage.

Reply:
Fair, I'll revert the change to the StdTx constructor. But I think you mean that `StdSignDoc`, not `StdSignMsg`, should have `omitempty`, since `StdSignDoc` is what is actually used when signing IIUC.